### ISSUE TITLE
[FR] Webhooks when moving files from pending -> downloaded

### ISSUE THREAD

**Issue by justcallmelarry** [kieraneglin/pinchflat#214]
I use https://github.com/Cloudbox/autoscan in order to not have my Plex library scan my entire disk every time there are changes to the file system.

So I was wondering if it would be possible to create some sort of webhook (at least) when things move from pending to downloaded (though re-downloads + deletes would also be great)?

It doesn't have to be exactly according to autoscan specification, but if it would include the normal info about the media + the path where it's downloaded I could at least adapt the information to send to autoscan myself

**Comment by kieraneglin** [comment]
Hey there! Thanks for the (additional) report ;)

I like this, but I'm actually working on a different system that should allow for this sort of thing as well! So I'm going to close this issue but I'll tag you in the PR with my other fix and you can let me know if it solves the problem you're having 🤙

**Comment by kieraneglin** [comment]
@justcallmelarry mentioned you in #219 - let me know if you think that'd work for you!

**Comment by kieraneglin** [comment]
@justcallmelarry still working on updating the docs, but the release with this change has been pushed and will be available in ~15 minutes 🤙 

Let me know how it works out!

**Comment by justcallmelarry** [comment]
@kieraneglin was away for the weekend, but set up my scripts today and it works splendidly for my use case 👌 

The only wish would be to be able say in settings or something which is the lifecycle script, so that i could volume it another way, but since i do daily backups i shouldn't lose the script anyways…

big thanks for the prompt responses on these issues and for following up on them even when they are closed! 🌟

**Comment by justcallmelarry** [comment]
if anyone finds this thread because searching for autoscan in some way, here is the script i settled on for now:
```sh
#!/bin/bash

EVENT_TYPE=$1
EVENT_DATA=$2

media_path=$(echo $EVENT_DATA | jq -r '.media_filepath')
media_dir=$(dirname "${media_path}")

handled_types=("media_downloaded", "media_deleted")

if ! [[ ${handled_types[@]} =~ $EVENT_TYPE ]]
then
  exit 0
fi

# note that you need to use the manual trigger since there is no pinchflat trigger in autoscan as of now
# --get -X POST looks weird but you want to specify it like this in order to send query params instead of form data
curl http://autoscan:3030/triggers/manual --data-urlencode "dir=${media_dir}" --get -X POST
```

**Comment by kieraneglin** [comment]
I like the idea, but me being assertive about the file location greatly simplifies the feature itself, the potential service area for bugs, and ease of setup (read: fewer help requests). All the good stuff! 

That said, you might be able to do what you want with symbolic links or custom volumes without needing to change any app configuration 🤙🏻

**Comment by g4m3r7ag** [comment]
Have you tried autopulse at all? I recently switched to that since autoscan is no longer maintained and I was starting to get errors in the logs about missing data when it would get requests from Sonarr. I'm trying to adapt your script to curl against autopulse but the format is slightly different and I'm not sure how I should format it. Here is an example curl from the autopulse documentation

curl -u 'admin:password' 'http://localhost:2875/triggers/manual?path=/path/to/file&hash=1234567890' 

As far as I can tell if you're not expecting the file to match a specific hash that can be omitted so I would assume

curl -u 'admin:password' 'http://localhost:2875/triggers/manual?path=/path/to/file'

You also build your own manual triggers so I created one called pinchflat

Here is the autopulse config for the trigger

triggers:
  pinchflat:
    type: manual
    rewrite:
      from: "/downloads"
      to: "/mnt/media/Youtube"

I then adapted the curl command as 

curl -u 'autopulseuser:autopulsepass' 'http://localhost:2875/triggers/pinchflat?path='"${media_dir}"

But unfortunately haven't been able to test it as I'm currently having problems getting pinchflat to use exported cookies.
